server/context.go

```diff
@@ -131,6 +134,37 @@ type Context struct {
 	TimeUntilStoreDead time.Duration
 }
 
+func getDefaultCacheSize() int64 {
+	mem := sigar.Mem{}
+	err := mem.Get()
+	if err != nil {
+		log.Infof("system available memory info is invalid, setting rocksdb cache size to 512MB %s", err)
+		return 512 << 20
+	}
+	sysAvlMem := int64(mem.Total)
+	if sysAvlMem < 0 {
```

Comment:
does this actually happen? And if so, it certainly should include `<= 0`. But it seems that if `mem.Get()` doesn't return an error, we should be able to trust `mem.Total` (?).